edx_when/migrations/0008_courseversion_block_type.py

```diff
@@ -0,0 +1,41 @@
+# Generated by Django 3.2.6 on 2021-08-31 15:11
+
+import opaque_keys.edx.django.models
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('edx_when', '0007_meta_tweaks'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='contentdate',
+            name='block_type',
+            field=models.CharField(max_length=255, null=True),
+        ),
+        migrations.AlterField(
+            model_name='contentdate',
+            name='course_id',
+            field=opaque_keys.edx.django.models.CourseKeyField(max_length=255),
+        ),
+        migrations.AlterField(
+            model_name='contentdate',
+            name='location',
+            field=opaque_keys.edx.django.models.UsageKeyField(default=None, max_length=255, null=True),
+        ),
+        migrations.AddIndex(
+            model_name='contentdate',
+            index=models.Index(fields=['course_id', 'block_type'], name='edx_when_course_block_type_idx'),
+        ),
+        migrations.AddIndex(
+            model_name='contentdate',
+            index=models.Index(fields=['course_id'], name='edx_when_course_id_idx'),
+        ),
+        migrations.AddIndex(
+            model_name='contentdate',
+            index=models.Index(fields=['location'], name='edx_when_location_idx'),
+        ),
+    ]
```

Comment:
Here's the SQL which this migration will run - generated with `./manage.py cms sqlmigrate edx_when 0008`:
```
BEGIN;
--
-- Add field block_type to contentdate
--
ALTER TABLE `edx_when_contentdate` ADD COLUMN `block_type` varchar(255) NULL;
--
-- Alter field course_id on contentdate
--
DROP INDEX `edx_when_contentdate_course_id_e6c39fdc` ON `edx_when_contentdate`;
--
-- Alter field location on contentdate
--
DROP INDEX `edx_when_contentdate_location_485206ea` ON `edx_when_contentdate`;
--
-- Create index edx_when_course_block_type_idx on field(s) course_id, block_type of model contentdate
--
CREATE INDEX `edx_when_course_block_type_idx` ON `edx_when_contentdate` (`course_id`, `block_type`);
--
-- Create index edx_when_course_id_idx on field(s) course_id of model contentdate
--
CREATE INDEX `edx_when_course_id_idx` ON `edx_when_contentdate` (`course_id`);
--
-- Create index edx_when_location_idx on field(s) location of model contentdate
--
CREATE INDEX `edx_when_location_idx` ON `edx_when_contentdate` (`location`);
COMMIT;
```

Reply:
Ah whoops. I didn't mean for additional SQL to be executed via the switch to models.Index vs the db_index parameter. Happy to have you keep it, but if you'd rather not keep in these additional changes, totally understand.

Reply:
I'm actually going to remove it - just to avoid risk by doing less index creation and dropping for *this* PR. Thanks!

Reply:
I've removed the index changes as mentioned - the SQL that will be executed is below:
```
BEGIN;
--
-- Add field block_type to contentdate
--
ALTER TABLE `edx_when_contentdate` ADD COLUMN `block_type` varchar(255) NULL;
--
-- Create index edx_when_course_block_type_idx on field(s) course_id, block_type of model contentdate
--
CREATE INDEX `edx_when_course_block_type_idx` ON `edx_when_contentdate` (`course_id`, `block_type`);
COMMIT;
```